code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -1253,14 +1135,21 @@ func (sa *StorageAllocation) payChallengePoolPassPaymentsToRemoveBlobber(sp *sta
 }
 
 func (sa *StorageAllocation) payCancellationCharge(sps []*stakePool, balances chainstate.StateContextI, passRates []float64, conf *Config, sc *StorageSmartContract, t *transaction.Transaction) error {
+	allocCost, err := sa.cost()
+	if err != nil {
+		return fmt.Errorf("failed to get allocation cost: %v", err)
+	}
+
 	cancellationCharge, err := sa.cancellationCharge(conf.CancellationCharge)
 	if err != nil {
 		return fmt.Errorf("failed to get cancellation charge: %v", err)
 	}
 
-	if sa.WritePool < cancellationCharge {
-		cancellationCharge = sa.WritePool
-		logging.Logger.Error("insufficient funds, %v, for cancellation charge, %v. distributing the remaining write pool.")
+	if allocCost-cancellationCharge < sa.WritePool {
```

Comment:
should it be 
`allocation_cost < cancellationCharge`?

Reply:
Yes, my bad. Fixed.